src/metric.cc

```diff
@@ -0,0 +1,279 @@
+  try {
+    stub->SendCustomMetricsMessage(
+        &custom_metrics_msg, PYTHONSTUB_MetricRequestValue, shm_handle_);
+  }
+  catch (const PythonBackendException& pb_exception) {
+    throw PythonBackendException(
+        "Failed to get metric value: " + std::string(pb_exception.what()));
+  }
+
+  return custom_metrics_msg->value;
+}
+
+void
+Metric::Clear()
+{
+  // Need to check if the metric has been cleared before as the Clear()'
+  // function can be called from two different locations: when the metric family
+  // clears the 'metric_map_' and when the 'Metric' object goes out of
+  // scope/being deleted.
+  if (!is_cleared_) {
+    is_cleared_ = true;
+    std::unique_ptr<Stub>& stub = Stub::GetOrCreateInstance();
+    SaveToSharedMemory(stub->ShmPool());
+    CustomMetricsMessage* custom_metrics_msg = nullptr;
+    try {
+      stub->SendCustomMetricsMessage(
+          &custom_metrics_msg, PYTHONSTUB_MetricRequestDelete, shm_handle_);
+    }
+    catch (const PythonBackendException& pb_exception) {
+      std::cerr << "Error when deleting Metric: " << pb_exception.what()
+                << "\n";
+    }
+  }
+}
+
+#else
+void*
+Metric::InitializeTritonMetric()
+{
+  std::vector<const TRITONSERVER_Parameter*> labels_params;
+  ParseLabels(labels_params, labels_);
+  TRITONSERVER_Metric* triton_metric = nullptr;
+  THROW_IF_TRITON_ERROR(TRITONSERVER_MetricNew(
+      &triton_metric,
+      reinterpret_cast<TRITONSERVER_MetricFamily*>(metric_family_address_),
+      labels_params.data(), labels_params.size()));
+  for (const auto label : labels_params) {
+    TRITONSERVER_ParameterDelete(const_cast<TRITONSERVER_Parameter*>(label));
+  }
+  return reinterpret_cast<void*>(triton_metric);
+}
+
+void
+Metric::ParseLabels(
+    std::vector<const TRITONSERVER_Parameter*>& labels_params,
+    const std::string& labels)
+{
+  triton::common::TritonJson::Value labels_json;
+  THROW_IF_TRITON_ERROR(labels_json.Parse(labels));
+
+  std::vector<std::string> members;
+  labels_json.Members(&members);
+  for (const auto& member : members) {
+    std::string value;
+    THROW_IF_TRITON_ERROR(labels_json.MemberAsString(member.c_str(), &value));
+    labels_params.emplace_back(TRITONSERVER_ParameterNew(
+        member.c_str(), TRITONSERVER_PARAMETER_STRING, value.c_str()));
+  }
+}
+
+void
+Metric::HandleMetricOperation(
+    CustomMetricsMessage* metrics_message_ptr,
+    const PYTHONSTUB_CommandType& command_type)
+{
+  if (command_type == PYTHONSTUB_MetricRequestValue) {
+    metrics_message_ptr->value = GetValue();
+  } else if (command_type == PYTHONSTUB_MetricRequestIncrement) {
+    Increment(operation_value_);
+  } else if (command_type == PYTHONSTUB_MetricRequestSet) {
+    SetValue(operation_value_);
+  } else {
+    throw PythonBackendException("Unknown metric operation");
+  }
+}
+
+void
+Metric::Increment(const double& value)
+{
+  auto triton_metric = reinterpret_cast<TRITONSERVER_Metric*>(metric_address_);
```

Comment:
Does it make sense to just store the `TRITONSERVER_Metric*`  as a member variable along side the metric_address after initialized?